### PR TITLE
File permissions

### LINKED ISSUE #628: Files: permissions
For e-times, implement the Files-REST permission factory to take into account the field `open_access`:
  1. create a new action to download files
  1. when trying to access bucket/file, fetch the record, fetch the field `open_access`
  2. is the field is `false`, then check if the user has the specific `Need/Action`

### PR DESCRIPTION
**Pull request by equadon**
* Permissions for files
* Removed language multi-field mapping for series
* UI fixes from comments during demo

Closes #628 

TODOs:
1. The test to download a file with an anonymous user now has a `sleep` otherwise it fails
2. Sometimes after uploading a file to an E-Item, the document is re-indexed but not with the last file uploaded

### CODE REVIEW

File: invenio_app_ils/permissions.py [hunk 1]
```diff
@@ -35,6 +37,22 @@ def backoffice_permission(*args, **kwargs):
     return Permission(backoffice_access_action)
 
 
+def file_download_permission(obj):
+    """File download permissions."""
+    search = current_app_ils.eitem_search
+    for hit in search.search_by_bucket_id(str(obj.bucket_id)).scan():
+        if not hit.to_dict().get("open_access", False):
```
Comment (zzacharo) [inline invenio_app_ils/permissions.py]: @equadon This should be checked against ES6, no?

File: ui/src/pages/backoffice/EItem/EItemDetails/components/EItemFiles/EItemFiles.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    this.maxFiles = invenioConfig.eitems.maxFiles;
+  }
+
+  onDelete = row => {
+    this.props.deleteFile(row.bucket, row.key);
+  };
+
+  onSelectFile = file => {
+    if (this.props.files.length >= this.maxFiles) {
+      this.props.addNotification(
```
Comment (zzacharo) [inline ui/src/pages/backoffice/EItem/EItemDetails/components/EItemFiles/EItemFiles.js]: @equadon nit: I am doing also something similar in the [oauth PR](https://github.com/inveniosoftware/invenio-app-ils/pull/638/files#diff-df7011847aeaa5e03ddbfc1cfa49b48cR12) if you like to adopt :)

File: ui/src/pages/backoffice/components/buttons/DownloadButton/DownloadButton.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    return (
+      <Button
+        icon
+        primary
+        as="a"
+        size="small"
+        labelPosition="left"
+        href={to}
+        {...(disabled ? { disabled: true } : {})}
+        {...(fluid ? { fluid: true } : {})}
```
Comment (zzacharo) [inline ui/src/pages/backoffice/components/buttons/DownloadButton/DownloadButton.js]: Why not just `<Button fluid={fluid ? true : false}`? Do I miss something?

File: ui/src/pages/backoffice/components/buttons/DownloadButton/DownloadButton.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+DownloadButton.propTypes = {
+  text: PropTypes.string,
+  fluid: PropTypes.bool,
+  disabled: PropTypes.bool,
+};
+
+DownloadButton.defaultProps = {
+  text: 'Download',
+  fluid: false,
```
Comment (zzacharo) [inline ui/src/pages/backoffice/components/buttons/DownloadButton/DownloadButton.js]: Actually you don't even need the check as you have already a default prop for the `fluid` property ;)

File: invenio_app_ils/permissions.py (hunk 1 above)
Comment (equadon) [reply invenio_app_ils/permissions.py]: The hit object didn't change much AFAIK from 6 to 7 so it should be okay but we'll see when the ES6 tests are running.

File: ui/src/pages/backoffice/EItem/EItemDetails/components/EItemFiles/EItemFiles.js (hunk 2 above)
Comment (equadon) [reply ui/src/pages/backoffice/EItem/EItemDetails/components/EItemFiles/EItemFiles.js]: You approach is cleaner, I'll use that too. :)

File: invenio_app_ils/permissions.py (hunk 1 above)
Comment (equadon) [reply invenio_app_ils/permissions.py]: I tested it both in ES6 and ES7 and `to_dict()` is supported but I'm fine with using the DB approach instead.

Comment (equadon) [reply invenio_app_ils/permissions.py]: I just remembered that we don't use Records-Files and only rely on Files-REST so we can't do this.

Comment (ntarocco) [reply invenio_app_ils/permissions.py]: True! Let's change it as discussed together :)

**Review by ntarocco — APPROVED** [review]

**Pull request merged**